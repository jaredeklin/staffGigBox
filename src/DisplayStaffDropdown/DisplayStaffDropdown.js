import React from 'react';
import { Api } from '../Api/Api';

const api = new Api();

const DisplayStaffDropdown = ({ staffRole, availableStaff, handleChange }) => {
  const role = api.rolesRegex(staffRole);
  const available = availableStaff.filter(staff => staff[role]);

  const staff = available.map(person => {
    return (
      <option key={person.staff_id} value={person.staff_id}>
        {person.name}
      </option>
    );
  });

  return (
    <select onChange={handleChange}>
      <option>Select staff</option>
      {staff}
    </select>
  );
};

export default DisplayStaffDropdown;
