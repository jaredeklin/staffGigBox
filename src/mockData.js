export const mockStaff = [
  {
    staff_id: 5,
    name: 'TK',
    bartender: true,
    barback: false,
    bar_manager: true,
    ass_bar_manager: true,
    beer_bucket: false,
    google_id: '1pLSNLEaSVT7hvRa5q4Vyy37IIN1'
  },
  {
    staff_id: 2,
    name: 'Jared',
    bartender: true,
    barback: false,
    bar_manager: false,
    ass_bar_manager: true,
    beer_bucket: false,
    google_id: '1pLSNLEaSVT7hvRa5q4Vyy37IIN2'
  },
  {
    staff_id: 3,
    name: 'Jesse',
    bartender: true,
    barback: false,
    bar_manager: false,
    ass_bar_manager: false,
    beer_bucket: false,
    google_id: '1pLSNLEaSVT7hvRa5q4Vyy37IIN1'
  },
  {
    staff_id: 4,
    name: 'Ross',
    bartender: false,
    barback: true,
    bar_manager: false,
    ass_bar_manager: false,
    beer_bucket: false,
    google_id: '1pLSNLEaSVT7hvRa5q4Vyy37IIN2'
  }
];

export const expectedStaff = [
  {
    id: 5,
    name: 'TK',
    bartender: true,
    barback: false,
    bar_manager: true,
    ass_bar_manager: true,
    beer_bucket: false,
    google_id: '1pLSNLEaSVT7hvRa5q4Vyy37IIN1'
  }
];

export const expectedStaffRoles = [
  { event_id: 3, role: 'Bar Manager', staff_id: 5 },
  { event_id: 3, role: 'Assistant Bar Manager', staff_id: 2 },
  { event_id: 3, role: 'Bartender', staff_id: 3 },
  { event_id: 3, role: 'Barback', staff_id: 4 }
];

export const mockUnscheduleStaff = [
  { event_id: 3, role: 'Bar Manager', staff_id: null, name: 'Staff Needed' },
  {
    event_id: 3,
    role: 'Assistant Bar Manager',
    staff_id: null,
    name: 'Staff Needed'
  },
  { event_id: 3, role: 'Bartender', staff_id: null, name: 'Staff Needed' },
  { event_id: 3, role: 'Barback', staff_id: null, name: 'Staff Needed' }
];

export const mockEmptySchedule = [
  {
    event_id: 3,
    role: 'Bar Manager',
    staff_id: null,
    event_date: '2018-07-20T07:00:00.000Z',
    schedule_id: 1
  },
  {
    event_id: 3,
    role: 'Assistant Bar Manager',
    staff_id: null,
    event_date: '2018-07-20T07:00:00.000Z',
    schedule_id: 2
  },
  {
    event_id: 3,
    role: 'Bartender',
    staff_id: null,
    event_date: '2018-07-20T07:00:00.000Z',
    schedule_id: 3
  },
  {
    event_id: 3,
    role: 'Barback',
    staff_id: null,
    event_date: '2018-07-20T07:00:00.000Z',
    schedule_id: 5
  }
];

export const mockEmptyScheduleWName = [
  {
    event_id: 3,
    role: 'Bar Manager',
    staff_id: null,
    name: 'Staff Needed',
    event_date: '2018-07-20T07:00:00.000Z',
    schedule_id: 1
  },
  {
    event_id: 3,
    role: 'Assistant Bar Manager',
    staff_id: null,
    name: 'Staff Needed',
    event_date: '2018-07-20T07:00:00.000Z',
    schedule_id: 2
  },
  {
    event_id: 3,
    role: 'Bartender',
    staff_id: null,
    name: 'Staff Needed',
    event_date: '2018-07-20T07:00:00.000Z',
    schedule_id: 3
  },
  {
    event_id: 3,
    role: 'Barback',
    staff_id: null,
    name: 'Staff Needed',
    event_date: '2018-07-20T07:00:00.000Z',
    schedule_id: 5
  }
];

export const mockEventInfo = {
  ass_bar_manager: true,
  bar_manager: true,
  barbacks: 1,
  bartenders: 1,
  date: 'Jul 20, 2018',
  event_id: 3,
  name: 'Billy Prince Billy'
};

export const mockEvent2Info = {
  ass_bar_manager: true,
  bar_manager: true,
  barbacks: 1,
  bartenders: 1,
  date: 'Jul 20, 2018',
  event_id: 4,
  name: 'Not Billy Prince Billy'
};

export const mockUnscheduledEvent = {
  ass_bar_manager: true,
  bar_manager: true,
  barbacks: 1,
  bartenders: 1,
  date: 'Jul 20, 2018',
  event_id: 3,
  name: 'Billy Prince Billy',
  staff: mockEmptyScheduleWName
};

export const mockEventsInfo = [mockEventInfo, mockEvent2Info];

export const mockSchedule = [
  { ...mockEventInfo, staff: mockEmptyScheduleWName }
];

export const mockFEScheduleData = [{ ...mockEventInfo, staff: mockStaff }];

export const mockFillRolesReturn = [
  { event_id: 3, role: 'Bar Manager', staff_id: 5, name: 'TK' },
  {
    event_id: 3,
    role: 'Assistant Bar Manager',
    staff_id: 2,
    name: 'Jared'
  },
  { event_id: 3, role: 'Barback', staff_id: 4, name: 'Ross' },
  { event_id: 3, role: 'Bartender', staff_id: 3, name: 'Jesse' }
];
////////////////
export const mockUnscheduledEvents = [
  { ...mockEventInfo, staff: mockUnscheduleStaff },
  { ...mockEvent2Info, staff: mockUnscheduleStaff }
];

export const mockNewSchedule = [
  { ...mockEventInfo, staff: mockFillRolesReturn },
  { ...mockEvent2Info, staff: [] }
];

export const mockEventToFillRoles = [
  {
    ...mockEventInfo,
    staff: [
      {
        event_id: 3,
        role: 'Bar Manager',
        staff_id: null,
        name: 'Staff Needed'
      },
      {
        event_id: 3,
        role: 'Assistant Bar Manager',
        staff_id: null,
        name: 'Staff Needed'
      },
      { event_id: 3, role: 'Bartender', staff_id: 3, name: 'Jesse' },
      { event_id: 3, role: 'Barback', staff_id: null, name: 'Staff Needed' }
    ]
  }
];

export const mockBuildRolesReturn = [
  {
    staff_id: null,
    event_id: 24,
    event_date: '2018-06-30',
    role: 'Bar Manager'
  },
  {
    staff_id: null,
    event_id: 24,
    event_date: '2018-06-30',
    role: 'Assistant Bar Manager'
  },
  {
    staff_id: null,
    event_id: 24,
    event_date: '2018-06-30',
    role: 'Bartender'
  },
  {
    staff_id: null,
    event_id: 24,
    event_date: '2018-06-30',
    role: 'Bartender'
  },
  {
    staff_id: null,
    event_id: 24,
    event_date: '2018-06-30',
    role: 'Bartender'
  },
  {
    staff_id: null,
    event_id: 24,
    event_date: '2018-06-30',
    role: 'Bartender'
  },
  {
    staff_id: null,
    event_id: 24,
    event_date: '2018-06-30',
    role: 'Barback'
  },
  {
    staff_id: null,
    event_id: 24,
    event_date: '2018-06-30',
    role: 'Barback'
  }
];

export const mockPostScheduleReturn = [
  {
    staff_id: null,
    name: 'Staff Needed',
    event_id: 24,
    event_date: '2018-06-30',
    role: 'Bar Manager',
    schedule_id: 1
  },
  {
    staff_id: null,
    name: 'Staff Needed',
    event_id: 24,
    event_date: '2018-06-30',
    role: 'Assistant Bar Manager',
    schedule_id: 2
  },
  {
    staff_id: null,
    name: 'Staff Needed',
    event_id: 24,
    event_date: '2018-06-30',
    role: 'Bartender',
    schedule_id: 3
  },
  {
    staff_id: null,
    name: 'Staff Needed',
    event_id: 24,
    event_date: '2018-06-30',
    role: 'Bartender',
    schedule_id: 4
  },
  {
    staff_id: null,
    name: 'Staff Needed',
    event_id: 24,
    event_date: '2018-06-30',
    role: 'Bartender',
    schedule_id: 5
  },
  {
    staff_id: null,
    name: 'Staff Needed',
    event_id: 24,
    event_date: '2018-06-30',
    role: 'Bartender',
    schedule_id: 6
  },
  {
    staff_id: null,
    name: 'Staff Needed',
    event_id: 24,
    event_date: '2018-06-30',
    role: 'Barback',
    schedule_id: 7
  },
  {
    staff_id: null,
    name: 'Staff Needed',
    event_id: 24,
    event_date: '2018-06-30',
    role: 'Barback',
    schedule_id: 8
  }
];

export const mockApiSchedule = [
  { schedule_id: 1, event_id: 3, staff_id: 3, role: 'Bartender' },
  { schedule_id: 2, event_id: 3, staff_id: 4, role: 'Barback' },
  { schedule_id: 3, event_id: 3, staff_id: null, role: 'Bar Manager' },
  {
    schedule_id: 4,
    event_id: 3,
    staff_id: null,
    role: 'Assistant Bar Manager'
  },
  { schedule_id: 5, event_id: 4, staff_id: 3, role: 'Bartender' }
];

export const mockApiScheduleReturn = {
  schedule: [],
  unscheduledEvents: [
    {
      ass_bar_manager: true,
      bar_manager: true,
      barbacks: 1,
      bartenders: 1,
      date: 'Jul 20, 2018',
      event_id: 3,
      name: 'Billy Prince Billy',
      staff: [
        {
          staff_id: 3,
          name: 'Jesse',
          bartender: true,
          barback: false,
          bar_manager: false,
          ass_bar_manager: false,
          beer_bucket: false,
          google_id: '1pLSNLEaSVT7hvRa5q4Vyy37IIN1',
          event_id: 3,
          role: 'Bartender',
          schedule_id: 1
        },
        {
          staff_id: 4,
          name: 'Ross',
          bartender: false,
          barback: true,
          bar_manager: false,
          ass_bar_manager: false,
          beer_bucket: false,
          google_id: '1pLSNLEaSVT7hvRa5q4Vyy37IIN2',
          event_id: 3,
          role: 'Barback',
          schedule_id: 2
        },
        {
          staff_id: null,
          name: 'Staff Needed',
          event_id: 3,
          role: 'Bar Manager',
          schedule_id: 3
        },
        {
          staff_id: null,
          name: 'Staff Needed',
          event_id: 3,
          role: 'Assistant Bar Manager',
          schedule_id: 4
        }
      ]
    }
  ]
};
