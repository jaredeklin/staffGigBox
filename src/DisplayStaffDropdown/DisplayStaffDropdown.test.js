import React from 'react';
import { shallow } from 'enzyme';
import DisplayStaffDropdown from './DisplayStaffDropdown';
import { mockStaff } from '../mockData';

describe('DisplayStaffDropdown', () => {
  it('should match the snapshot', () => {
    const mockRole = 'Bar Manager';
    const mockAvailableStaff = mockStaff;
    const mockHandleChange = jest.fn();

    const wrapper = shallow(
      <DisplayStaffDropdown
        staffRole={mockRole}
        availableStaff={mockAvailableStaff}
        handleChange={mockHandleChange}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
