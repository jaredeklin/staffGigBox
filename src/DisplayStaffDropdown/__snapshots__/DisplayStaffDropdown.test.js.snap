// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplayStaffDropdown should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DisplayStaffDropdown
    availableStaff={
        Array [
            Object {
              "ass_bar_manager": true,
              "bar_manager": true,
              "barback": false,
              "bartender": true,
              "beer_bucket": false,
              "google_id": "1pLSNLEaSVT7hvRa5q4Vyy37IIN1",
              "name": "TK",
              "staff_id": 5,
            },
            Object {
              "ass_bar_manager": true,
              "bar_manager": false,
              "barback": false,
              "bartender": true,
              "beer_bucket": false,
              "google_id": "1pLSNLEaSVT7hvRa5q4Vyy37IIN2",
              "name": "Jared",
              "staff_id": 2,
            },
            Object {
              "ass_bar_manager": false,
              "bar_manager": false,
              "barback": false,
              "bartender": true,
              "beer_bucket": false,
              "google_id": "1pLSNLEaSVT7hvRa5q4Vyy37IIN1",
              "name": "Jesse",
              "staff_id": 3,
            },
            Object {
              "ass_bar_manager": false,
              "bar_manager": false,
              "barback": true,
              "bartender": false,
              "beer_bucket": false,
              "google_id": "1pLSNLEaSVT7hvRa5q4Vyy37IIN2",
              "name": "Ross",
              "staff_id": 4,
            },
          ]
    }
    handleChange={[Function]}
    staffRole="Bar Manager"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <option>
          Select staff
</option>,
        Array [
          <option
            value={5}
>
            TK
</option>,
        ],
      ],
      "onChange": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Select staff",
        },
        "ref": null,
        "rendered": "Select staff",
        "type": "option",
      },
      Object {
        "instance": null,
        "key": "5",
        "nodeType": "host",
        "props": Object {
          "children": "TK",
          "value": 5,
        },
        "ref": null,
        "rendered": "TK",
        "type": "option",
      },
    ],
    "type": "select",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <option>
            Select staff
</option>,
          Array [
            <option
              value={5}
>
              TK
</option>,
          ],
        ],
        "onChange": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Select staff",
          },
          "ref": null,
          "rendered": "Select staff",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": "5",
          "nodeType": "host",
          "props": Object {
            "children": "TK",
            "value": 5,
          },
          "ref": null,
          "rendered": "TK",
          "type": "option",
        },
      ],
      "type": "select",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
